packages/integration-sdk-core/src/data/createIntegrationEntity.ts

```diff
@@ -179,6 +179,52 @@ function generateEntity({
   return entity;
 }
 
+/**
+ * Validates that the provided value conforms to the supported types.
+ * If the value is an array, this function is called recursively on each element
+ * to ensure nested arrays are also validated. This function throws an error
+ * if it encounters a value type that is not supported.
+ *
+ * @param value The value to be validated. It can be a single value or an array.
+ *              For arrays, each element is validated recursively.
+ * @param path The path to the current property being validated, used for error messaging.
+ *             This is updated with each recursive call to reflect the current context.
+ */
+export function validateValueType(
+  value: any,
+  path: string,
+  depth: number = 1,
+): void {
+  // Explicitly allow null values
+  if (value === null) {
+    return;
+  }
+
+  if (Array.isArray(value)) {
+    // If the depth is > 1 then we won't allow arrays inside arrays.
+    if (depth > 1) {
+      throw new IntegrationError({
+        code: 'UNSUPPORTED_TYPE',
+        message: `Unsupported type found at "${path}": Nested arrays are not supported.`,
+      });
+    }
+
+    // If the value is an array, validate each element
+    value.forEach((item, index) => {
+      validateValueType(item, `${path}[${index}]`, ++depth);
+    });
+  } else {
+    // For non-array values, check if the type is supported
+    const valueType = typeof value;
+    if (!SUPPORTED_TYPES.includes(valueType)) {
+      throw new IntegrationError({
```

Comment:
How can we keep steps from failing on unsupported data? This is a step towards strictness that'll have any overall impact on job statuses and the ingestion team.
How else could we improve the overall health of jobs and data?

Reply:
We should open a bigger discussion for this but this should be a MAJOR, we have some ideas in the AWS Pod as to how to proceed